tests/integration/targets/rds_cluster_param_group/tasks/main.yaml

```diff
@@ -0,0 +1,255 @@
+
+    # Test adding invalid or not modifiable parameters
+    - name: Update existing RDS cluster parameter group with tags (idempotency)
+      amazon.aws.rds_cluster_param_group:
+        name: "{{ rds_cluster_param_group_name }}"
+        db_parameter_group_family: "{{ dbparam_group_family }}"
+        description: "RDS cluster param group for Engine {{ engine_versions.db_engine_versions.0.engine_version }}"
+        parameters:
+          archive_library: "testing_lib"  # This parameter is not modifiable
+          invalid_fake: true  # this parameter does not exist
+      register: update_parameters
+      ignore_errors: true
+
+    - name: Ensure module failed to update invalid/not modifiable parameters
+      assert:
+        that:
+          - update_parameters is failed
+          - update_parameters.msg == error_msg
+      vars:
+        error_msg: "The following parameters are not modifiable: archive_library. The following parameters are not available parameters for the 'postgres16' DB parameter group family: invalid_fake."
+
+    # Test adding parameters
+    - name: Modify RDS cluster parameter group with new parameters
+      amazon.aws.rds_cluster_param_group:
+        name: "{{ rds_cluster_param_group_name }}"
+        db_parameter_group_family: "{{ dbparam_group_family }}"
+        description: "RDS cluster param group for Engine {{ engine_versions.db_engine_versions.0.engine_version }}"
+        parameters:
+          array_nulls: 0
+          authentication_timeout: 50
+      register: update_parameters
+
+    - name: Ensure module failed to update invalid/not modifiable parameters
+      assert:
+        that:
+          - update_parameters is changed
+
+    # Test adding parameters (idempotency)
+    - name: Modify RDS cluster parameter group with new parameters (idempotency)
+      amazon.aws.rds_cluster_param_group:
+        name: "{{ rds_cluster_param_group_name }}"
+        db_parameter_group_family: "{{ dbparam_group_family }}"
+        description: "RDS cluster param group for Engine {{ engine_versions.db_engine_versions.0.engine_version }}"
+        parameters:
+          array_nulls: 0
+          authentication_timeout: 50
+      register: update_idempotency
+
+    - name: Ensure module did not reported change
+      assert:
+        that:
+          - update_idempotency is not changed
+
+    # Test update existing parameter value
+    - name: Modify RDS cluster parameter group existing parameter value
+      amazon.aws.rds_cluster_param_group:
+        name: "{{ rds_cluster_param_group_name }}"
+        db_parameter_group_family: "{{ dbparam_group_family }}"
+        description: "RDS cluster param group for Engine {{ engine_versions.db_engine_versions.0.engine_version }}"
+        parameters:
+          array_nulls: 0
+          authentication_timeout: 100
+      register: update_parameters
+
+    - name: Ensure module reported change
+      assert:
+        that:
+          - update_parameters is changed
```

Comment:
Could we include a check to verify the updated parameter value?

Reply:
unfortunately, there is no API to retrieve the RDS cluster param group parameter value

Reply:
Please correct me if my understanding is wrong. I see the response of [RDS.Client.describe_db_cluster_parameters ](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds/client/describe_db_cluster_parameters.html#describe-db-cluster-parameters), contains the parameter name and value.

Reply:
updated